astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -2082,6 +2097,70 @@ def test_unstructured_to_structured(self):
         # For the other tests of ``structured_to_unstructured``, see
         # ``test_structured.TestStructuredQuantityFunctions.test_unstructured_to_structured``
 
+    def test_merge_arrays(self):
+        """Test `numpy.lib.recfunctions.merge_arrays`."""
+        # merge 1 normal array
+        arr = rfn.merge_arrays(self.q_pv["p"])
+        assert np.array_equal(arr.value["f0"], [1, 2, 3])
+        assert arr.unit == (u.km,)
+
+        # merge 1 structured array
+        arr = rfn.merge_arrays(self.q_pv)
+        assert np.array_equal(arr.value["p"], [1, 2, 3])
+        assert np.array_equal(arr.value["v"], [0.25, 0.5, 0.75])
+        assert arr.unit == (u.km, u.km/u.s)
+
+        # merge 1-elt tuple
+        arr = rfn.merge_arrays((self.q_pv,))
+        assert np.array_equal(arr.value["p"], [1, 2, 3])
+        assert np.array_equal(arr.value["v"], [0.25, 0.5, 0.75])
+        assert arr.unit == (u.km, u.km/u.s)
+
+        # merge 2-elt tuples
+        arr = rfn.merge_arrays((self.q_pv, self.q_pv_t))
+        assert np.array_equal(arr.value["f0"]["p"], [1, 2, 3])
```

Comment:
My tendency would be to simplify the tests, e.g., here doing `assert_array_equal(arr['f0'], self.q_pc)` - i.e., rely on `assert_array_equal` being sufficiently tested (if only everywhere else in this test function!).